heron/common/src/cpp/network/client.cpp

```diff
@@ -218,3 +218,8 @@ void Client::StartBackPressureConnectionCb(Connection* conn) {
 void Client::StopBackPressureConnectionCb(Connection* conn) {
   // Nothing to be done here. Should be handled by inheritors if they care about backpressure
 }
+
+bool Client::HasCausedBackPressure() const {
+  Connection* conn = static_cast<Connection*>(conn_);
```

Comment:
Do we need to check state&pointer here? Can conn_ be NULL in some cases/states?

Reply:
Not really, but I've added some checks just for safety